src/tinbu-parser.js

```
@@ -63,6 +88,9 @@ function extractObservation( xmlDoc, isMetric ) {
         xmlDoc, null, XPathResult.FIRST_ORDERED_NODE_TYPE )
       .singleNodeValue;
   }
+  if ( observation === null ) {
+    throw new Error( "Observation data is missing" );
+  }
```

@ezequielc do you think we can live with either a missing `observation` or `location`? ie. maybe some locations don't return forecasts, etc. Thanks.

---

@alex-deaconu Good question, I am not sure. I didn't see any case like that. Given that we don't have documentation, I would prefer to throw the error for now so it is easier to debug in case we get that case. Thoughts? Thanks

---

Makes sense. Thanks!